plat/kvm/arm/exceptions.S

```diff
@@ -112,18 +113,27 @@
 	 */
 	mrs	x0, tpidr_el1
 	ldr	x0, [x0, #LCPU_AUXSP_OFFSET]
+	/* NOTE: We should normally align the stack before doing this
+	 * subtraction because we must ensure that the `ectx` field
+	 * is aligned to the corresponding ECTX alignment.
+	 * However, this is guaranteed to already be the case for the
+	 * auxiliary stack because it is allocated with this exact alignment
+	 * in mind.
+	 */
+	sub	x0, x0, #UK_SYSCALL_CTX_SIZE
 
 	/* We now have in SP the trap stack and in x0 the auxiliary stack */
 	EXCHANGE_SP_WITH_X0  /* Switch them */
 	/* Restore old SP we stored before system call check */
 	ldr	x0, [x0, #-16]
-	str	x0, [sp, #-16]  /* Store old SP in auxiliary stack */
+	str	x0, [sp, #__SP_OFFSET]  /* Store old SP in auxiliary stack */
+	b	1f
 0:
 	/* Restore x0 */
```

Comment:
Why not restore x0 in the case of a system call

Reply:
Yeah you're right, x0 should have been restored on syscalls as well of course. I must have somehow wrongly changed this during the review process and not notice. The restoration of x0 should have been placed after the `1:` label instead.

Reply:
@mogasergiu , would this require an update (PR) from your side?

Reply:
Yes I could look into it after I return from my vacation, unless someone else would like to have a go at it as well. Though the fix should be rather simple: moving the restoration of x0 two lines below should suffice if I see this right.

Reply:
Done #1256